remix-analyzer/src/solidity-analyzer/modules/staticAnalysisCommon.js

```diff
@@ -517,6 +517,15 @@ function isDynamicArrayAccess (node) {
   return node && nodeType(node, exactMatch(nodeTypes.IDENTIFIER)) && (node.attributes.type.endsWith('[] storage ref') || node.attributes.type === 'bytes storage ref' || node.attributes.type === 'string storage ref')
 }
 
+/**
+ * True if node accesses 'length' member of array
+ * @node {ASTNode} node to check for
+ * @return {bool}
+ */
+function isDynamicArrayLengthAccess (node) {
+  return node && nodeType(node, exactMatch(nodeTypes.MEMBERACCESS)) && (node.attributes.member_name === 'length') && node.children[0].attributes.type.indexOf('[]') !== -1
```

Comment:
Same here, create intermediary variables. Else it's going to be super difficult for the next person to take over your code.

Reply:
sounds reasonable.